tpu-provisioner/internal/cloud/gke.go

```diff
@@ -80,11 +82,21 @@ func (g *GKE) EnsureNodePoolForPod(p *corev1.Pod, why string) error {
 	// "do: googleapi: Error 400: Cluster is running incompatible operation ..."
 	// To avoid a bunch of failed requests, we dedeuplicate here.
 	if _, inProgress := g.inProgressCreates.Load(name); inProgress {
-		return ErrDuplicateRequest
+		return fmt.Errorf("creation ongoing for node pool name: %v: %w", name, ErrDuplicateRequest)
 	}
 	g.inProgressCreates.Store(name, struct{}{})
 	defer g.inProgressCreates.Delete(name)
 
+	// A restarting JobSet will trigger a new Node Pool creation.
+	// This creation might overlap with the previous one, so we need to deduplicate.
+	if jobKey := p.Labels[jobKeyLabel]; jobKey != "" {
```

Comment:
Can we add to the comment here, explaining the jobKey is stable through restarts but the job UID (used for NP name) is not, so that is why we need to dedupe on the job key